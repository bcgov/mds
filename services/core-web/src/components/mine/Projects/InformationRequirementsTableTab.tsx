import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import { Tabs } from "antd";
import { formatDate } from "@mds/common/redux/utils/helpers";
import {
  fetchRequirements,
  fetchProjectById,
  updateInformationRequirementsTable,
} from "@mds/common/redux/actionCreators/projectActionCreator";
import { getInformationRequirementsTableStatusCodesHash } from "@mds/common/redux/selectors/staticContentSelectors";
import {
  getProject,
  getInformationRequirementsTable,
  getRequirements,
} from "@mds/common/redux/selectors/projectSelectors";
import ReviewInformationRequirementsTable from "@/components/mine/Projects/ReviewInformationRequirementsTable";
import UpdateInformationRequirementsTableForm from "@/components/Forms/informationRequirementsTable/UpdateInformationRequirementsTableForm";

const sideMenuOptions = [
  {
    href: "intro-project-overview",
    title: "Introduction and Project Overview",
  },
  {
    href: "baseline-information",
    title: "Baseline Information",
  },
  {
    href: "mine-plan",
    title: "Mine Plan",
  },
  {
    href: "reclamation-closure-plan",
    title: "Reclamation Closure Plan",
  },
  {
    href: "modelling-mitigation-discharges",
    title: "Modelling Mitigation Discharges",
  },
  {
    href: "environmental-assessment-predictions",
    title: "Environmental Assessment Predictions",
  },
  {
    href: "environmental-monitoring",
    title: "Environmental Monitory",
  },
  {
    href: "health-safety",
    title: "Health Safety",
  },
  {
    href: "management-plan",
    title: "Management Plan",
  },
];

const InformationRequirementsTableTab = () => {
  const { projectGuid } = useParams<{ projectGuid: string }>();
  const dispatch = useDispatch();

  const project = useSelector(getProject);
  const requirements = useSelector(getRequirements);
  const informationRequirementsTable = useSelector(getInformationRequirementsTable);
  const informationRequirementsTableStatusCodesHash = useSelector(
    getInformationRequirementsTableStatusCodesHash
  );

  const [isLoaded, setIsLoaded] = useState(false);

  const handleFetchData = () => {
    dispatch(fetchProjectById(projectGuid))
      .then(() => dispatch(fetchRequirements()))
      .catch(setIsLoaded(false));
  };

  useEffect(() => {
    handleFetchData();
  }, []);

  useEffect(() => {
    if (
      projectGuid &&
      project.project_guid === projectGuid &&
      informationRequirementsTable?.project_guid === projectGuid
    ) {
      setIsLoaded(true);
    }
  }, [project, informationRequirementsTable]);

  const deepMergeById = (r1, r2) =>
    r1.map(({ requirement_guid, sub_requirements, ...rest }) => ({
      requirement_guid,
      ...rest,
      ...r2.find((i) => i.requirement_guid === requirement_guid),
      sub_requirements: deepMergeById(sub_requirements, r2),
    }));

  const handleUpdateIRT = (event, values) => {
    event.preventDefault();
    dispatch(
      updateInformationRequirementsTable(
        { projectGuid, informationRequirementsTableGuid: informationRequirementsTable.irt_guid },
        values
      )
    ).then(() => handleFetchData());
  };

  const { information_requirements_table } = project;
  const updateUser = information_requirements_table?.update_user;
  const updateDate = formatDate(information_requirements_table?.update_timestamp);
  const statusCode = information_requirements_table?.status_code;

  const mergedRequirements = deepMergeById(
    requirements,
    informationRequirementsTable?.requirements?.filter(({ deleted_ind }) => deleted_ind === false)
  );

  return (
    <div>
      <Tabs
        size="large"
        defaultActiveKey={sideMenuOptions[0].href}
        className="now-tabs vertical-tabs"
        tabPosition="left"
        animated={{ inkBar: true, tabPane: false }}
        items={sideMenuOptions.map((tab, idx) => ({
          key: tab.href,
          label: tab.title,
          children: (
            <div>
              <UpdateInformationRequirementsTableForm
                initialValues={{
                  status_code: statusCode,
                }}
                displayValues={{
                  statusCode,
                  updateUser,
                  updateDate,
                  informationRequirementsTableStatusCodesHash: informationRequirementsTableStatusCodesHash,
                }}
                handleSubmit={handleUpdateIRT}
              />
              <ReviewInformationRequirementsTable
                requirements={mergedRequirements[idx]}
                isLoaded={isLoaded}
              />
            </div>
          ),
        }))}
      />
    </div>
  );
};
export default InformationRequirementsTableTab;
