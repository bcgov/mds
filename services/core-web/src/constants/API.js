import queryString from "query-string";

// Document Generation: Notices of Work

export const GET_NOTICE_OF_WORK_APPLICATION_DOCUMENT_CONTEXT_TEMPLATE = (
  documentTypeCode,
  context_guid
) =>
  `/now-applications/application-document-types/${documentTypeCode}?${queryString.stringify({
    context_guid,
  })}`;
export const RETRIEVE_CORE_DOCUMENT = (token) => `/documents?${queryString.stringify(token)}`;
