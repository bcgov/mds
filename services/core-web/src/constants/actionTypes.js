export const DOCUMENTS = "DOCUMENTS";
export const STORE_DOCUMENT_CONTEXT_TEMPLATE = "STORE_DOCUMENT_CONTEXT_TEMPLATE";
