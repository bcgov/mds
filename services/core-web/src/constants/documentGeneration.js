import { currencyMask } from "@common/utils/helpers";
import { required, number, currency } from "@common/utils/Validate";

export const CURRENCY = [currencyMask, { validate: [required, number, currency] }];

export const CONTEXT_PROPS = { currency: CURRENCY };

export const getContextProps = (contextProps) => {
  let props = {};
  if (contextProps) {
    contextProps.map((contextProp) =>
      CONTEXT_PROPS[contextProp].map((prop) => (props = { ...props, ...prop }))
    );
  }
  return props;
};
