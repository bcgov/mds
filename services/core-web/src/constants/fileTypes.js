/* eslint-disable no-dupe-keys */
export const EXCEL = {
  ".xls": "application/vnd.ms-excel",
  ".xlsx": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
};

export const PDF = {
  ".pdf": "application/pdf",
};

export const DOC = {
  ".doc": "application/msword",
  ".docx": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
  ".rtf": "application/rtf",
};

export const OPEN_DOC = {
  ".odt": "application/vnd.oasis.opendocument.text",
  ".ott": "application/vnd.oasis.opendocument.text-template",
  ".oth": "application/vnd.oasis.opendocument.text-web",
  ".odm": "application/vnd.oasis.opendocument.text-master",
};

export const DOCUMENT = { ...PDF, ...DOC, ...OPEN_DOC };

export const UNIQUELY_SPATIAL = {
  ".dbf": "application/dbf",
  ".geoJSon": "application/vnd.geo+json",
  ".gml": "application/gml+xml",
  ".kml": "application/vnd.google-earth.kml+xml ",
  ".kmz": "application/vnd.google-earth.kmz",
  ".prj": "application/octet-stream",
  ".sbn": "application/octet-stream",
  ".sbx": "application/octet-stream",
  ".shp": "application/octet-stream",
  ".shpz": "application/octet-stream",
  ".shx": "application/octet-stream",
  ".wkt": "application/octet-stream",
};

export const SPATIAL = { ...UNIQUELY_SPATIAL, ".csv": "text/csv", ".xml": "application/xml" };
