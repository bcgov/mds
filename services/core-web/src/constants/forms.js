// Redux Forms
export const ADD_FULL_PARTY = "ADD_FULL_PARTY";
export const ADD_ROLES = "ADD_ROLES";
export const ADD_QUICK_PARTY = "ADD_QUICK_PARTY";
export const ADD_TENURE_NUMBER = "ADD_TENURE_NUMBER";
export const MINE_RECORD = "MINE_RECORD";
export const SEARCH_COORDINATES = "SEARCH_COORDINATES";
export const UPDATE_PERMITTEE = "UPDATE_PERMITTEE";
export const ADD_TAILINGS = "ADD_TAILINGS";
export const ADD_TAILINGS_REPORT = "ADD_TAILINGS_REPORT";
export const EDIT_TAILINGS = "EDIT_TAILINGS";
export const ADD_PARTY_RELATIONSHIP = "ADD_PARTY_RELATIONSHIP";
export const EDIT_PARTY_RELATIONSHIP = "EDIT_PARTY_RELATIONSHIP";
export const MINE_ADVANCED_SEARCH = "MINE_ADVANCED_SEARCH";
export const CONTACT_ADVANCED_SEARCH = "CONTACT_ADVANCE_SEARCH";
export const VARIANCE_ADVANCED_SEARCH = "VARIANCE_ADVANCED_SEARCH";
export const INCIDENT_ADVANCED_SEARCH = "INCIDENT_ADVANCED_SEARCH";
export const REPORT_ADVANCED_SEARCH = "REPORT_ADVANCED_SEARCH";
export const ADD_MINESPACE_USER = "ADD_MINESPACE_USER";
export const ADD_PERMIT = "ADD_PERMIT";
export const PERMIT_AMENDMENT = "PERMIT_AMENDMENT";
export const EDIT_PERMIT = "EDIT_PERMIT";
export const EDIT_FULL_PARTY = "EDIT_FULL_PARTY";
export const MINE_COMPLIANCE_FILTER = "MINE_COMPLIANCE_FILTER";
export const ADD_VARIANCE = "ADD_VARIANCE";
export const EDIT_VARIANCE = "EDIT_VARIANCE";
export const MINE_INCIDENT = "MINE_INCIDENT";
export const ADD_REPORT = "ADD_REPORT";
export const EDIT_REPORT = "EDIT_REPORT";
export const FILTER_REPORTS = "FILTER_REPORTS";
export const GENERATE_DOCUMENT = "GENERATE_DOCUMENT";

// Notice Of Work
export const NOTICE_OF_WORK_SEARCH = "NOTICE_OF_WORK_SEARCH";
export const EDIT_NOTICE_OF_WORK = "EDIT_NOTICE_OF_WORK";
export const EDIT_NOTICE_OF_WORK_DOCUMENT_FORM = "EDIT_NOTICE_OF_WORK_DOCUMENT_FORM";
export const CHANGE_NOW_MINE = "CHANGE_NOW_MINE";
export const UPDATE_NOW_LEAD_INSPECTOR = "UPDATE_NOW_LEAD_INSPECTOR";
export const ADD_NOW_REVIEW = "ADD_NOW_REVIEW";
export const MM_PERMIT_APPLICATION_CREATE = "MM_PERMIT_APPLICATION_CREATE";
export const CHANGE_NOW_LOCATION = "CHANGE_NOW_LOCATION";
export const MAJOR_MINE_PERMIT_APPLICATION_CREATE = "MAJOR_MINE_PERMIT_APPLICATION_CREATE";
