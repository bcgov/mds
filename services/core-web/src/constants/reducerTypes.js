// Notices of Work
export const GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT =
  "GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT";
