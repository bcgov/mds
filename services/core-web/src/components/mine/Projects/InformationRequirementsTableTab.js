import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Tabs, Tag } from "antd";
import "@ant-design/compatible/assets/index.css";
import PropTypes from "prop-types";
import { formatDate } from "@common/utils/helpers";
import {
  fetchRequirements,
  fetchProjectById,
  updateInformationRequirementsTable,
} from "@common/actionCreators/projectActionCreator";
import { getInformationRequirementsTableStatusCodesHash } from "@common/selectors/staticContentSelectors";
import {
  getProject,
  getInformationRequirementsTable,
  getRequirements,
} from "@common/selectors/projectSelectors";
import { ArrowLeftOutlined, EnvironmentOutlined } from "@ant-design/icons";
import ReviewInformationRequirementsTable from "@/components/mine/Projects/ReviewInformationRequirementsTable";
import UpdateInformationRequirementsTableForm from "@/components/Forms/informationRequirementsTable/UpdateInformationRequirementsTableForm";
import CustomPropTypes from "@/customPropTypes";
import * as routes from "@/constants/routes";

const propTypes = {
  project: CustomPropTypes.project.isRequired,
  requirements: PropTypes.arrayOf(CustomPropTypes.requirements).isRequired,
  informationRequirementsTable: CustomPropTypes.informationRequirementsTable.isRequired,
  informationRequirementsTableStatusCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectGuid: PropTypes.string,
      irtGuid: PropTypes.string,
    }),
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  fetchRequirements: PropTypes.func.isRequired,
  updateInformationRequirementsTable: PropTypes.func.isRequired,
};

const sideMenuOptions = [
  {
    href: "intro-project-overview",
    title: "Introduction and Project Overview",
  },
  {
    href: "baseline-information",
    title: "Baseline Information",
  },
  {
    href: "mine-plan",
    title: "Mine Plan",
  },
  {
    href: "reclamation-closure-plan",
    title: "Reclamation Closure Plan",
  },
  {
    href: "modelling-mitigation-discharges",
    title: "Modelling Mitigation Discharges",
  },
  {
    href: "environmental-assessment-predictions",
    title: "Environmental Assessment Predictions",
  },
  {
    href: "environmental-monitoring",
    title: "Environmental Monitory",
  },
  {
    href: "health-safety",
    title: "Health Safety",
  },
  {
    href: "management-plan",
    title: "Management Plan",
  },
];

export class InformationRequirementsTableTab extends Component {
  mergedRequirements = [];

  constructor(props) {
    super(props);
    this.state = {
      fixedTop: false,
      activeTab: "intro-project-overview",
    };
  }

  componentDidMount() {
    this.handleFetchData();
    window.addEventListener("scroll", this.handleScroll);
    this.handleScroll();
  }

  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }

  /* eslint-disable no-param-reassign */
  deepMergeById = (r1, r2) =>
    r1.map(({ requirement_guid, sub_requirements, ...rest }) => ({
      requirement_guid,
      ...rest,
      ...r2.find((i) => i.requirement_guid === requirement_guid),
      sub_requirements: this.deepMergeById(sub_requirements, r2),
    }));

  handleScroll = () => {
    if (window.pageYOffset > 170 && !this.state.fixedTop) {
      this.setState({ fixedTop: true });
    } else if (window.pageYOffset <= 170 && this.state.fixedTop) {
      this.setState({ fixedTop: false });
    }
  };

  handleTabChange = (activeTab) => {
    const { projectGuid, irtGuid } = this.props.match.params;
    const url = routes.INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(projectGuid, irtGuid, activeTab);
    this.props.history.push(url);
    this.setState({ activeTab });
  };

  handleFetchData = () => {
    const { projectGuid } = this.props.match.params;
    if (projectGuid) {
      return this.props
        .fetchProjectById(projectGuid)
        .then(() => this.props.fetchRequirements())
        .then(() => this.setState({ isLoaded: true }))
        .catch(() => this.setState({ isLoaded: false }));
    }
    return null;
  };

  handleUpdateIRT = (event, values) => {
    event.preventDefault();
    const { projectGuid, irtGuid } = this.props.match.params;
    return this.props
      .updateInformationRequirementsTable(
        {
          projectGuid,
          informationRequirementsTableGuid: irtGuid,
        },
        values
      )
      .then(() => this.handleFetchData());
  };

  render() {
    const {
      project_title,
      mine_name,
      mine_guid,
      project_guid,
      information_requirements_table,
    } = this.props.project;
    const updateUser = information_requirements_table?.update_user;
    const updateDate = formatDate(information_requirements_table?.update_timestamp);
    const statusCode = information_requirements_table?.status_code;

    this.mergedRequirements = this.deepMergeById(
      this.props.requirements,
      this.props.informationRequirementsTable.requirements
    );

    return (
      <div className="page">
        <div
          className={
            this.state.fixedTop
              ? "padding-lg view--header fixed-scroll"
              : " padding-lg view--header"
          }
        >
          <h1>
            {project_title}
            <span className="padding-sm--left">
              <Tag title={`Mine: ${mine_name}`}>
                <Link
                  style={{ textDecoration: "none" }}
                  to={routes.MINE_GENERAL.dynamicRoute(mine_guid)}
                  disabled={!mine_guid}
                >
                  <EnvironmentOutlined className="padding-sm--right" />
                  {mine_name}
                </Link>
              </Tag>
            </span>
          </h1>
          <Link to={routes.PROJECTS.dynamicRoute(project_guid)}>
            <ArrowLeftOutlined className="padding-sm--right" />
            Back to: {mine_name} Project overview
          </Link>
          <Tabs
            size="large"
            animated={{ inkBar: true, tabPane: false }}
            className="now-tabs"
            style={{ margin: "0" }}
            centered
          >
            <Tabs.TabPane tab="IRT" key="irt">
              <></>
            </Tabs.TabPane>
          </Tabs>
        </div>
        <div>
          <Tabs
            size="large"
            activeKey={this.state.activeTab}
            className="now-tabs vertical-tabs"
            tabPosition="left"
            animated={{ inkBar: true, tabPane: false }}
            onChange={(tab) => this.handleTabChange(tab)}
          >
            {sideMenuOptions.map((tab, idx) => (
              <Tabs.TabPane tab={tab.title} key={tab.href} className="vertical-tabs--tabpane">
                <div>
                  <UpdateInformationRequirementsTableForm
                    initialValues={{
                      status_code: statusCode,
                    }}
                    displayValues={{
                      statusCode,
                      updateUser,
                      updateDate,
                      informationRequirementsTableStatusCodesHash: this.props
                        .informationRequirementsTableStatusCodesHash,
                    }}
                    handleSubmit={this.handleUpdateIRT}
                  />
                  <ReviewInformationRequirementsTable
                    requirements={this.mergedRequirements[idx]}
                    isLoaded={this.state.isLoaded}
                  />
                </div>
              </Tabs.TabPane>
            ))}
          </Tabs>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  project: getProject(state),
  informationRequirementsTable: getInformationRequirementsTable(state),
  requirements: getRequirements(state),
  informationRequirementsTableStatusCodesHash: getInformationRequirementsTableStatusCodesHash(
    state
  ),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchProjectById,
      fetchRequirements,
      updateInformationRequirementsTable,
    },
    dispatch
  );

InformationRequirementsTableTab.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(InformationRequirementsTableTab);
