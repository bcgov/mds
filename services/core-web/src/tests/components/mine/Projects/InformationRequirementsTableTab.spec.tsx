import React from "react";
import { render } from "@testing-library/react";
import InformationRequirementsTableTab from "@/components/mine/Projects/InformationRequirementsTableTab";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@/tests/utils/ReduxWrapper";
import { PROJECTS } from "@mds/common/constants/reducerTypes";

const initialState = {
  [PROJECTS]: {
    projects: MOCK.PROJECTS.records,
    project: MOCK.PROJECT,
    informationRequirementsTable: MOCK.INFORMATION_REQUIREMENTS_TABLE,
    requirements: MOCK.REQUIREMENTS.records,
  },
};

function mockFunction() {
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    useParams: jest.fn().mockReturnValue({
      projectGuid: "35633148-57f8-4967-be35-7f89abfbd02e",
    }),
    useHistory: jest.fn().mockReturnValue({
      push: jest.fn(),
    }),
  };
}

jest.mock("react-router-dom", () => mockFunction());

describe("InformationRequirementsTableTab", () => {
  it("renders properly", () => {
    const { container } = render(
      <ReduxWrapper initialState={initialState}>
        <InformationRequirementsTableTab />
      </ReduxWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
