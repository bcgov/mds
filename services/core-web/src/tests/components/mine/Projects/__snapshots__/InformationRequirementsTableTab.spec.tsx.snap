// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationRequirementsTableTab renders properly 1`] = `
<div>
  <div
    class="ant-tabs ant-tabs-left ant-tabs-large now-tabs vertical-tabs"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-tab ant-tabs-tab-active"
            data-node-key="intro-project-overview"
          >
            <div
              aria-controls="rc-tabs-test-panel-intro-project-overview"
              aria-selected="true"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-intro-project-overview"
              role="tab"
              tabindex="0"
            >
              Introduction and Project Overview
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="baseline-information"
          >
            <div
              aria-controls="rc-tabs-test-panel-baseline-information"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-baseline-information"
              role="tab"
              tabindex="0"
            >
              Baseline Information
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="mine-plan"
          >
            <div
              aria-controls="rc-tabs-test-panel-mine-plan"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-mine-plan"
              role="tab"
              tabindex="0"
            >
              Mine Plan
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="reclamation-closure-plan"
          >
            <div
              aria-controls="rc-tabs-test-panel-reclamation-closure-plan"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-reclamation-closure-plan"
              role="tab"
              tabindex="0"
            >
              Reclamation Closure Plan
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="modelling-mitigation-discharges"
          >
            <div
              aria-controls="rc-tabs-test-panel-modelling-mitigation-discharges"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-modelling-mitigation-discharges"
              role="tab"
              tabindex="0"
            >
              Modelling Mitigation Discharges
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="environmental-assessment-predictions"
          >
            <div
              aria-controls="rc-tabs-test-panel-environmental-assessment-predictions"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-environmental-assessment-predictions"
              role="tab"
              tabindex="0"
            >
              Environmental Assessment Predictions
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="environmental-monitoring"
          >
            <div
              aria-controls="rc-tabs-test-panel-environmental-monitoring"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-environmental-monitoring"
              role="tab"
              tabindex="0"
            >
              Environmental Monitory
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="health-safety"
          >
            <div
              aria-controls="rc-tabs-test-panel-health-safety"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-health-safety"
              role="tab"
              tabindex="0"
            >
              Health Safety
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="management-plan"
          >
            <div
              aria-controls="rc-tabs-test-panel-management-plan"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-management-plan"
              role="tab"
              tabindex="0"
            >
              Management Plan
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-left"
      >
        <div
          aria-hidden="false"
          aria-labelledby="rc-tabs-test-tab-intro-project-overview"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          id="rc-tabs-test-panel-intro-project-overview"
          role="tabpanel"
          tabindex="0"
        >
          <div>
            <form
              class="ant-legacy-form ant-legacy-form-vertical"
            >
              <div
                class="ant-col ant-col-24"
              >
                <div
                  class="ant-alert ant-alert-warning ant-alert-with-description"
                  data-show="true"
                  role="alert"
                >
                  <span
                    aria-label="exclamation-circle"
                    class="anticon anticon-exclamation-circle ant-alert-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="exclamation-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
                      />
                    </svg>
                  </span>
                  <div
                    class="ant-alert-content"
                  >
                    <div
                      class="ant-alert-message"
                    >
                      N/A
                    </div>
                    <div
                      class="ant-alert-description"
                    >
                      <div
                        class="ant-row"
                      >
                        <div
                          class="ant-col ant-col-xs-24 ant-col-md-18"
                        >
                          <p>
                            Final IRT was submitted by 
                             on
                             
                            , waiting for ministry review.
                          </p>
                          <b>
                            Please note that updating this status will notify the project proponent.
                          </b>
                        </div>
                        <div
                          class="ant-col ant-col-xs-24 ant-col-md-6"
                        >
                          <div
                            class="ant-row ant-legacy-form-item"
                          >
                            <div
                              class="ant-col ant-legacy-form-item-control-wrapper"
                            >
                              <div
                                class="ant-legacy-form-item-control"
                              >
                                <span
                                  class="ant-legacy-form-item-children"
                                >
                                  <div
                                    class="ant-row ant-legacy-form-item"
                                  >
                                    <div
                                      class="ant-col ant-legacy-form-item-control-wrapper"
                                    >
                                      <div
                                        class="ant-legacy-form-item-control"
                                      >
                                        <span
                                          class="ant-legacy-form-item-children"
                                        >
                                          <div
                                            class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                                          >
                                            <div
                                              class="ant-select-selector"
                                            >
                                              <span
                                                class="ant-select-selection-search"
                                              >
                                                <input
                                                  aria-activedescendant="status_code_list_0"
                                                  aria-autocomplete="list"
                                                  aria-controls="status_code_list"
                                                  aria-haspopup="listbox"
                                                  aria-owns="status_code_list"
                                                  autocomplete="off"
                                                  class="ant-select-selection-search-input"
                                                  id="status_code"
                                                  role="combobox"
                                                  type="search"
                                                  value=""
                                                />
                                              </span>
                                              <span
                                                class="ant-select-selection-placeholder"
                                              >
                                                Action
                                              </span>
                                            </div>
                                            <span
                                              aria-hidden="true"
                                              class="ant-select-arrow"
                                              style="user-select: none;"
                                              unselectable="on"
                                            >
                                              <span
                                                aria-label="down"
                                                class="anticon anticon-down ant-select-suffix"
                                                role="img"
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  data-icon="down"
                                                  fill="currentColor"
                                                  focusable="false"
                                                  height="1em"
                                                  viewBox="64 64 896 896"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                                  />
                                                </svg>
                                              </span>
                                            </span>
                                          </div>
                                        </span>
                                      </div>
                                    </div>
                                  </div>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <div
              id="loading-screen"
            >
              <div
                aria-busy="true"
                aria-live="polite"
                class="ant-spin ant-spin-spinning"
              >
                <span
                  aria-label="loading"
                  class="anticon anticon-loading loader loader-color ant-spin-dot"
                  role="img"
                  style="font-size: 170px; margin-top: 70px;"
                >
                  <svg
                    aria-hidden="true"
                    class="anticon-spin"
                    data-icon="loading"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
